### PR TITLE
Update Heroku stack to heroku-20 and bump Ruby version to 2.5.9

### PR DESCRIPTION
**Pull request by matyikriszta**
### Description
We need to update the stack we are using on Heroku. We are currently on heroku-18 which reached EOL in April. This means we can no longer deploy to Heroku. This PR updates the stack to heroku-20 and the Ruby version to 2.5.9 which is the earliest supported Ruby on the heroku-20 stack.

### Status
Ready for Review

### Related Github issue
#1876

### CODE REVIEW

**Review by KimberleyCook — APPROVED** [review]

**Comment by gnclmorais** [comment]
Sweet work! I guess this PR needs to be reviewed and merged before https://github.com/codebar/planner/pull/1874…

File: native-installation-instructions.md [hunk 1]
```diff
@@ -16,19 +16,19 @@ These are the original instructions for natively installing the app to your mach
...
 ## Set up a Ruby Environment
 
-You will need to install Ruby 2.4.2 using RVM or rbenv.
+You will need to install Ruby 2.7.2 using RVM or rbenv.
 
 ### Option 1: Using [rvm](https://rvm.io/rvm/install)
 
 ```bash
-rvm install 2.4.2
+CPPFLAGS=-DUSE_FFI_CLOSURE_ALLOC rvm install 2.5.9
```
Comment (gnclmorais) [inline native-installation-instructions.md:24]: Could we add a comment explaining this flag?

**Review by gnclmorais — APPROVED** [review]

File: native-installation-instructions.md (hunk 1 above)
Comment (matyikriszta) [reply native-installation-instructions.md:24]: @gnclmorais sure, although we are not going to stay on this version of Ruby for long, now that deploy is unblocked we are going to start upgrading Ruby/Rails

**Review by BPScott — APPROVED** [review]
Paired with Kirszta on this at the time.

Old-ruby on M1 macs requires that funky flag to cause a successful install, but that shall go away once we get onto ruby 3.1/3.2.

**Pull request merged**